pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-jwt.adoc

```diff
@@ -0,0 +1,660 @@
+pub fn new(algorithm: SigningAlgorithm, key_length: SigningKeyLength, key: String) -> Result<SignatureValidator, JWTError>
+
+----
+
+PDK supports the following signing algorithms and key lengths:
+
+[%autowidth.spread,cols="a,a"]
+|===
+| Signing Algorithm | Signing Key Length
+| RSA | 256, 384 and 512
+| HMAC | 256, 384 and 512
+| ES | 256 and 384
+|===
+
+The `SigningAlgorithm` enum defines the available signing algorithms:
+
+[source,Rust]
+----
+pub enum SigningAlgorithm {
+    Rsa,
+    Hmac,
+    Es,
+}
+----
+
+The `SigningKeyLength` enum defines the available signing key lengths:
+
+[source,Rust]
+----
+pub enum SigningKeyLength {
+    Len256,
+    Len384,
+    Len512,
+}
+----
+
+A `SignatureValidator` instance implements the `SignatureValidation` trait. This `SignatureValidation` trait exposes a `validate` function that accepts tokens as `String` parameters and returns a `Result` that contains the parsed `JWTClaims` or a `JWTError` in the event of an error:
+
+[source,Rust]
+----
+pub trait SignatureValidation {
+    fn validate(&self, token: String) -> Result<JWTClaims, JWTError>;
+}
+----
+
+The following code example shows how to initialize the the `SignatureValidator` and validate an incoming token signed with a 256 byte HMAC algorithm:
+
+[source,Rust]
+----
+use anyhow::Result;
+use pdk::hl::*;
+use pdk::jwt::*;
+use pdk::script::HandlerAttributesBinding;
+// [...]
+
+async fn filter(
+    state: RequestState,
+    signature_validator: &SignatureValidator,
+    token_provider: &TokenProvider,
+) -> Flow<()> {
+    let headers_state = state.into_headers_state().await;
+
+    // Extract token
+    let token = token_provider.provide(&HandlerAttributesBinding::partial(headers_state.handler()));
+
+    if token.is_err() {
+        return Flow::Break(Response::new(401).with_body("Bearer not found"));
+    }
+
+    // Validating signature
+    let claims = signature_validator.validate(token.unwrap());
+
+    if claims.is_err() {
+        return Flow::Break(Response::new(401).with_body("Invalid token"));
+    }
+    // [...]
+}
+
+#[entrypoint]
+async fn configure(launcher: Launcher, Configuration(configuration): Configuration) -> Result<()> {
+    let config: Config = serde_json::from_slice(&configuration)?;
+
+    let signature_validator = SignatureValidator::new(
+        model::SigningAlgorithm::Hmac,
+        model::SigningKeyLength::Len256,
+        config.clone().secret,
+    )?;
+
+    let token_provider = TokenProvider::CustomProvider { expression: config.token_extractor.clone() };
+
+    launcher
+        .launch(on_request(|request| {
+            filter(request, &signature_validator, &token_provider)
+        }))
+        .await?;
+    Ok(())
+}
+----
+
+Because, the `SignatureValidator` confirms that the policy configurations is correct during initialization, initializing the `SignatureValidator` in the `#[entrypoint]` function throws an early error when the policy is applied rather than when the API receives a request.
```

Comment:
```suggestion
Because, the `SignatureValidator` confirms that the policy configurations are correct during initialization, initializing the `SignatureValidator` in the `#[entrypoint]` function throws an early error when the policy is applied rather than when the API receives a request.
```